src/components/Cell/Cell.tsx

```diff
@@ -59,23 +57,27 @@ const Cell = ({ id, garden, setGarden, waterGarden, bullDoze, setBullDoze, filte
         setIsPopulated(true);
         setIsPlanted(true);
       }
-
-      setCell(foundCell);
+      if (foundCell && foundCell.updated_at) {
+        let updateValue = convertTime(lastUpdate[foundCell.location_id]) > convertTime(foundCell.updated_at) ? lastUpdate[foundCell.location_id] : foundCell.updated_at;
+        handleUpdate(foundCell.location_id, updateValue);
+      }
     }
     // eslint-disable-next-line
   }, []);
 
   useEffect(() => {
-    if (isPlanted && cellContents?.plant.watering && cellContents?.plant.watering !== 'None' && typeof cellContents?.plant.updated_at !== 'undefined') {
-      const today = Date.now() + new Date(cellContents?.plant.updated_at).getTimezoneOffset() * 60 * 1000
-      const daysToAdd = WATERING_SCHEDULE[cellContents?.plant.watering]
-      const nextWatering = new Date(cellContents?.plant.updated_at).getTime() + (daysToAdd * 24 * 60 * 60 * 1000)
-      console.log(cellContents?.plant.updated_at)
-      console.log(today, nextWatering)
-      if (today > nextWatering) {
+    if (isPlanted && cellContents && cellContents.plant.watering && cellContents.plant.watering !== 'None' && typeof cellContents.plant.updated_at !== 'undefined') {
+      let now = Date.now() + (new Date().getTimezoneOffset() * 60 * 1000);
+      let lastUpdateAdjusted = convertTime(lastUpdate[cellContents.plant.location_id]);
+      let interval = WATERING_SCHEDULE[cellContents?.plant.watering] * 24 * 60 * 60 * 1000;
+      if ((now - lastUpdateAdjusted) >= interval) {
         setNeedsWatering(true);
       }
+      setTimeout(() => {
+        setNeedsWatering(true);
+      }, interval - (now-lastUpdateAdjusted));
```

Comment:
Super awesome you were able to wrap all of this up in a setTimeout!!  Did you spend any time researching for packages that could have eliminated a lot of the heavy lifting you did below handling/converting time?